test/interval_tests/construction.jl

```diff
@@ -340,3 +342,16 @@ import IntervalArithmetic: force_interval
     @test force_interval(Inf, -Inf) == Interval(-Inf, Inf)
     @test_throws ArgumentError force_interval(NaN, 3)
 end
+
+@testset "#320 fixing convert bug" begin
+    x = 3..4
+    T = typeof(x)
+
+    m = @which convert(T, x)
+
+    @test m.sig == Tuple{typeof(convert),Type{Interval{T}},Interval{T}} where T
+
+    x = interval(1//1, 5//3)
+    @test convert(typeof(x), x) == x
```

Comment:
```suggestion
    @test convert(typeof(x), x) === x
```
This would check that it does not create a new interval.